test/inspect.test.js

```diff
@@ -1,27 +1,194 @@
+    },
+    msg: 'django looks ok',
+  },
+  jinja2: {
+    data: {
+      name: 'jinja2',
+      version: '2.7.2',
+      dependencies: {
+        markupsafe: {
+          name: 'markupsafe',
+          version: /.+$/,
+        },
+      },
+    },
+    msg: 'jinja2 looks ok',
+  },
+  'python-etcd': {
+    data: {
+      name: 'python-etcd',
+      version: '0.4.5',
+      dependencies: {
+        dnspython: {
+          name: 'dnspython',
+          version: /.+$/,
+        },
+        urllib3: {
+          name: 'urllib3',
+          version: /.+$/,
+        },
+      },
+    },
+    msg: 'python-etcd is ok',
+  },
+  'django-select2': {
+    data: {
+      name: 'django-select2',
+      version: '6.0.1',
+      dependencies: {
+        'django-appconf': {
+          name: 'django-appconf',
+        },
+      },
+    },
+    msg: 'django-select2 looks ok',
+  },
+  irc: {
+    data: {
+      name: 'irc',
+      version: '16.2',
+      dependencies: {
+        'more-itertools': {},
+        'jaraco.functools': {},
+        'jaraco.collections': {
+          dependencies: {
+            'jaraco.text': {},
+          }
+        },
+        'jaraco.text': {
+          dependencies: {
+            'jaraco.collections': {}
+          }
+        },
+      }
+    },
+    msg: 'irc ok, even though it has a cyclic dep, yay!',
+  },
+  testtools: {
+    data: {
+      name: 'testtools',
+      version: '2.3.0',
+      dependencies: {
+        'pbr': {},
+        'extras': {},
+        'fixtures': {},
+        'unittest2': {},
+        'traceback2': {},
+        'python-mimeparse': {},
+      }
+    },
+    msg: 'testtools ok, even though it\'s cyclic, yay!',
+  },
+};
+
 test('inspect', function (t) {
-  chdirWorkspaces('pip-app');
+  return Promise.resolve().then(function () {
+    chdirWorkspaces('pip-app');
+    t.teardown(testUtils.activateVirtualenv('pip-app'));
+  })
+  .then(function () {
+    return plugin.inspect('.', 'requirements.txt')
+  })
+  .then(function (result) {
+    var plugin = result.plugin;
+    var pkg = result.package;
+
+    t.test('plugin', function (t) {
+      t.ok(plugin, 'plugin');
+      t.equal(plugin.name, 'snyk-python-plugin', 'name');
+      t.match(plugin.runtime, 'Python', 'runtime');
+      t.end();
+    });
 
-  return plugin.inspect('.', 'requirements.txt')
+    t.test('package', function (t) {
+      t.ok(pkg, 'package');
+      t.equal(pkg.name, 'pip-app', 'name');
+      t.equal(pkg.version, '0.0.0', 'version');
+      t.end();
+    });
+
+    t.test('package dependencies', function (t) {
+      Object.keys(pipAppExpectedDependencies).forEach(function (depName) {
+        t.match(
+          pkg.dependencies[depName],
+          pipAppExpectedDependencies[depName].data,
+          pipAppExpectedDependencies[depName].msg
+        );
+      });
+
+      t.end();
+    });
+
+    t.end();
+  });
+});
+
+test('transitive dep not installed', function (t) {
+  return Promise.resolve().then(function () {
```

Comment:
The non-async calls could have still stayed outside the Promise chain I think? but it's ok anyhow